BUG: optimize failure on wheels

Master branch wheel builds have [been failing](https://travis-ci.org/MacPython/scipy-wheels/builds/540546542) since around 18 days ago, with the traceback below the fold only observed for Python 3.5 / 3.7 on MacOS:

<details>

```
_______________________ TestCurveFit.test_broadcast_y _________________________

[gw0] darwin -- Python 3.5.4 /Users/travis/build/MacPython/scipy-wheels/venv/bin/python



self = <scipy.optimize.tests.test_minpack.TestCurveFit object at 0x118e37cc0>



    def test_broadcast_y(self):

        xdata = np.arange(10)

        target = 4.7 * xdata ** 2 + 3.5 * xdata + np.random.rand(len(xdata))

        fit_func = lambda x, a, b: a*x**2 + b*x - target

        for method in ['lm', 'trf', 'dogbox']:

            popt0, pcov0 = curve_fit(fit_func,

                                     xdata=xdata,

                                     ydata=np.zeros_like(xdata),

                                     method=method)

            popt1, pcov1 = curve_fit(fit_func,

                                     xdata=xdata,

                                     ydata=0,

                                     method=method)

>           assert_array_equal(pcov0, pcov1)

E           AssertionError: 

E           Arrays are not equal

E           

E           (mismatch 100.0%)

E            x: array([[ 0.00011 , -0.000783],

E                  [-0.000783,  0.005931]])

E            y: array([[ 0.00011 , -0.000783],

E                  [-0.000783,  0.005931]])



fit_func   = <function TestCurveFit.test_broadcast_y.<locals>.<lambda> at 0x118a48158>

method     = 'dogbox'

pcov0      = array([[ 0.00011024, -0.00078326],

       [-0.00078326,  0.00593076]])

pcov1      = array([[ 0.00011024, -0.00078326],

       [-0.00078326,  0.00593076]])

popt0      = array([ 4.67757254,  3.76916206])

popt1      = array([ 4.67757254,  3.76916206])

self       = <scipy.optimize.tests.test_minpack.TestCurveFit object at 0x118e37cc0>

target     = array([   0.4236548 ,    8.84589411,   26.23758721,   53.691773  ,

         90.16366276,  135.38344152,  190.99172504,  255.32889492,

        329.36804456,  413.12559664])

xdata      = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])



../venv/lib/python3.5/site-packages/scipy/optimize/tests/test_minpack.py:787: AssertionError

```
</details>

If someone does determine the cause of this failure, it would also be useful to know why it shows up in the wheel repo and not in dev CI, so that we can continue to work toward reducing the dissonance between the two.

--- a/scipy/optimize/tests/test_minpack.py
+++ b/scipy/optimize/tests/test_minpack.py
@@ -791,7 +791,7 @@ class TestCurveFit(object):
                                      xdata=xdata,
                                      ydata=0,
                                      method=method)
-            assert_allclose(pcov0, pcov1, rtol=1e-12)
+            assert_allclose(pcov0, pcov1)
 
 
 class TestFixedPoint(object):

